code/software/libs/src/start_serial/link_scripts/rosco_m68k_kernel.ld

```diff
@@ -16,16 +16,45 @@
 
 OUTPUT_FORMAT("binary")
 ENTRY(START)
-kram = 0x00001000;
-phys = 0x00F00000;      /* "high" address for pos-ind. init   */
+
+ram  = 0x00001000;          /* start of user memory                     */
+init = 0x01028000;          /* "pretend" high address for pos-ind. init */
+
 MEMORY
 {
-  INIT : org = phys, l = 0xBEFFF   /* init -> top of ram (767KB) */ 
-  KRAM : org = kram, l = 0xFEFFF   /* Allow located code to use all of RAM > 4k */
+  INIT : org = init, l = 0x00010000     /* 64K for init (reused)        */ 
+  RAM  : org = ram,  l = 0x00FFEFFF     /* program can use all RAM > 4K */
 }
 
-PROVIDE(_STACK_TOP   = 0x00100000);   /* default stack at 1MB base RAM top  */
-PROVIDE(_MFP_VECBASE = 0x00000040);   /* handy vector base, useable from C  */
+/* program configuration symbols that may be useful to override */
+/* using linker option: --defsym=<symbol_name>=<value>          */
+PROVIDE(_RAM_SIZE     = 1M);            /* default to 1MB of RAM        */
+PROVIDE(_STACK_TOP    = _RAM_SIZE);     /* default stack at top of RAM  */
+PROVIDE(_STACK_SIZE   = 0x00004000);    /* 16KB default stack size      */
+PROVIDE(_HEAP_END     = _STACK_TOP - _STACK_SIZE); /* heap end at stack */
+
+/* firware absolute addresses (generally not useful to override)  */
+PROVIDE(_MFP_VECTORS    = 0x00000100);  /* MFP interrupt vector base    */
+PROVIDE(_TIMER_100HZ    = 0x0000040C);  /* firmware 100Hz timer counter */
+PROVIDE(_EASY68K_ECHOON = 0x00000410);  /* Easy68k 'echo on' flag       */
+PROVIDE(_EASY68K_PROMPT = 0x00000411);  /* Easy68k 'prompt on' flag     */
+PROVIDE(_EASY68K_SHOWLF = 0x00000412);  /* Easy68k 'LF display' flag    */
+
+PROVIDE(_EFP_TABLE      = 0x00000420);  /* Extension function ptr table */
+
+PROVIDE(_EFP_PRINT      = 0x00000420);  /* Print string to console      */
```

Comment:
I like these :) 

Thinking out loud, is it worth adding a comment in the code that sets these up to remind that if the layout changes, this must also be changed? Maybe in `code/firmware/rosco_m68k_v1.2/equates.S` or `code/firmware/rosco_m68k_v1.2/bootstrap.S` (or both)?

Reply:
Yes, that sounds worth doing.  Not "ideal" to have to change things in sync multiple places, but not too many options here I am aware of (and for absolute symbols, linker script is "best" option - since can't directly declare them C).  I am mostly thinking these are kind of "bowels of system" (not directly C callable etc.), but a comment to remind of the "matched set" is always good.

Reply:
I agree. I can't think of a nice way to have a single source-of-truth for this value in this case.

Reply:
Comment added to keep in sync with firmware equates.S